### DISCUSSION TITLE
Updating the Wiki

### DISCUSSION THREAD

**Discussion by Woodhouse3D** [ThePix/QuestJS#109] [General]
The QuestJS wiki is nothing to sneeze at, but personally I found it much more difficult to navigate than the Quest 5 docs.
For instance, I've just started looking at all the pages in the pages dropdown menu... Seems like it has pages that aren't listed in the sidebar below it??? idk.

**Comment by ThePix** [comment]
I have gone through and hopefully added missing pages to the contents list.

I am worried about how easy it is to find stuff. I wrote so I know where to look, and that makes it tricky to put myself in the position of someone else. If you have a suggestion..?

**Reply by KVonGit** [reply]
My two cents:

The main thing that makes it difficult (for me) is the way the wikis are laid out.

Most of the time it's just me, though, I think. For instance, I went directly to [this](https://github.com/ThePix/QuestJS/wiki/Naming-Items-and-Rooms#other-attributes "https://github.com/ThePix/QuestJS/wiki/Naming-Items-and-Rooms#other-attributes") just now, but [yesterday I couldn't find how to stop an article from printing before a name that wasn't capitalized](https://github.com/ThePix/QuestJS/discussions/112 "https://github.com/ThePix/QuestJS/discussions/112") for the life of me.

(I also just noticed Pixie's reply in that thread.( NOTE TO SELF: Make sure my notifications are on, haha.))

---
How much work would it be to copy those wiki files into a `docs` directory, not unlike Quest 5's `docs` directory?

If that wouldn't be too much work, we could do something like that, and you could enable GitHub Pages using the `docs` directory. Then, `https://ThePix.github.io/QuestJS` could be your docs home page.

That's what http://docs.textadventures.co.uk is -- GH Pages, and it's using the optional subdomain.

(Does PixDocs generate the files like the ones in the Quest 5 docs?)

---
Or maybe just some GH Pages with links to the existing wiki pages might work even better.

You can do all sorts of stuff with GitHub Pages now. You can use MD files or HTML. I'm using GH Pages to host [the testing site](https://textmisadventures.github.io/quest/ "https://textmisadventures.github.io/quest/"). Here's that HTML file: https://github.com/TextMisadventures/quest/blob/gh-pages/index.html

It even lets us use jQuery and Bootstrap. 

---
It would be super-cool to have a QuestJS game as the docs landing page. Sadly, we can't run a QuestJS game on GH Pages "out-of-the-box." I tried last night, but I think it only failed to load files because of some QuestJS settings. I am running [a Quest 5 game](https://kvongit.github.io/quest/bony_king-563-html_version/) I converted to a [website](https://github.com/KVonGit/quest/tree/gh-pages/bony_king-563-html_version) on GH pages, and it's basically the same file structure and everything as a QuestJS game.

Here's my QuestJS fork with GH Pages enabled (and no file changes or additions): https://kvongit.github.io/QuestJS/

It loads **index.html** from the root directory by default when we enable GH Pages. So, it's trying to run the bare bones game, but it fails to load "lib/_util.js" and "lib/_settings.js".

![image](https://github.com/user-attachments/assets/d2918fcc-6a7c-4917-a31b-3a0ac28862c1)

**Comment by ThePix** [comment]
"How much work would it be to copy those wiki files into a docs directory, not unlike Quest 5's docs directory?"

The problem with that is that every update to the documentation is a new commit.

Also, I am not sure how the Quest docs work. Somewhere there must be a server that handles them, using the GitHub files as a source. Likely that is an Azure server - though I never worked out where the code is - which would cost money.

**Reply by KVonGit** [reply]
>  how the Quest docs work. Somewhere there must be a server that handles them, using the GitHub files as a source. 

It's GitHub Pages.

There's a "Pages" selection under Settings:
![image](https://github.com/user-attachments/assets/e555ecca-b4ac-4cb1-8407-9212ca5bcc7d)

---
We can select the [`docs` folder from the `master` (or main) branch](https://github.com/KVonGit/quest/tree/master/docs "https://github.com/KVonGit/quest/tree/master/docs"):
![image](https://github.com/user-attachments/assets/bba2d026-c272-4326-a280-33252b8612f3)

---
### EDIT
I just applied those settings in my `KVonGit/quest` fork, and I am now hosting the docs on my GH Pages site. Check it out: https://kvongit.github.io/quest/

---
We can also create a `gh-pages` branch to use specifically as the docs branch. (That's how the quest repo once did it, but it was changed to a docs folder in the main directory at some point.)

---
The only thing special is Alex has the subdomain `http://docs.textadventures.co.uk` set up to use with the Quest docs.

![image](https://github.com/user-attachments/assets/dca1dfb3-ee9c-4f17-842e-77b76096b0dd)

---
> The problem with that is that every update to the documentation is a new commit.

Oh, I see. I use git to update my wikis; so, I wasn't even thinking about this being an extra step. `:)`

Plus, this might mean having to revise multiple files (maybe, probably), which probably wouldn't be worth all the effort in the long run.

---
**EDIT**

As @FloridaChristopher says, though:
> The QuestJS wiki is nothing to sneeze at

Adding GH Pages to the mix is probably unwise at this juncture. Plus, the changes made to the wiki since yesterday have already improved the experience on my end. 👍🏼 👍🏼
